ssh-agent.c

```diff
@@ -375,13 +461,39 @@ process_sign_request2(SocketEntry *e)
 		verbose_f("%s key not found", sshkey_type(key));
 		goto send;
 	}
-	if (id->confirm && confirm_key(id) != 0) {
+	/*
+	 * If session IDs were recorded for this socket, then use them to
+	 * annotate the confirmation messages with the host keys.
+	 */
+	if (e->nsession_ids > 0 &&
+	    parse_userauth_request(data, key, &user, &sid) == 0) {
+		for (i = 0; i < e->nsession_ids; i++) {
+			if ((fp = sshkey_fingerprint(e->session_ids[i].key,
+			    SSH_FP_HASH_DEFAULT, SSH_FP_DEFAULT)) == NULL)
+				fatal_f("fingerprint failed");
+			if (buf_equal(sid, e->session_ids[i].sid) == 0) {
+				debug3_f("destination %s %s (slot %zu)",
+				    sshkey_type(e->session_ids[i].key), fp, i);
+				xasprintf(&sig_dest, "public key request for "
+				    "user \"%s\" to %s %s", user,
```

Comment:
it's a request for login as 'user' to host 'key' requested by 'fp', no?
so in this case it shout not say 'to %s' but "forwarded by %s' ? no?

Reply:
it's **as** user but **to** host